src/utils/png_reader.h

```diff
@@ -0,0 +1,9 @@
+#ifndef PNG_READER_H
+#define PNG_READER_H
+
+#include "video/vga_palette.h"
+#include <stdbool.h>
+
+bool png_read_paletted(const char *filename, unsigned char *dst);
```

Comment:
Perhaps your png_reader methods should return SD_ERRORCODE instead of boolean?
then we can have real errorcodes rather than just "General Error"

Reply:
We don't use these codes anywhere; hey were a leftover from when this stuff was in a separate library. TODO remove the old error code stuff and use abort() where necessary.